src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -9,18 +9,146 @@
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private Node<K, V>[] keyArray;
+    private int size;
+    private final static int MIN_ARRAY_CAPACITY = 16;
+    private int threshold;
+
+    public MyHashMap() {
+        this.keyArray = (Node<K, V>[]) new Node[MIN_ARRAY_CAPACITY];
+        this.size = size;
+        this.threshold = (int) (keyArray.length * 0.75);
+    }
+
     @Override
     public void put(K key, V value) {
-        
+        int arrayIndex = findIndex(key);
+        if (threshold == size) {
+            reSize();
+        }
+        addNewNode(key, value, arrayIndex);
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        int newArrayIndex = findIndex(key);
+        if (Objects.equals(keyArray[newArrayIndex].key, key)) {
+            return keyArray[newArrayIndex].value;
+        }
+        if (!Objects.equals(keyArray[newArrayIndex].next, null)) {
+            Node<K, V> nextKey = keyArray[newArrayIndex].next;
+            while (nextKey != null) {
+                if (Objects.equals(nextKey.key, key)) {
+                    return nextKey.value;
+                }
+                nextKey = nextKey.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean isEmpty(int arrayIndex) {
+        return Objects.equals(keyArray[arrayIndex], null);
+    }
+
+    private void reSize() {
+        int newArraySize = keyArray.length * 2;
+        Node<K, V>[] newKeyArray = (Node<K, V>[]) new Node[newArraySize];
+        threshold = (int) (newArraySize * 0.75);
+        size = 0;
+        Node<K, V>[] oldKeyArray = keyArray;
+        keyArray = newKeyArray;
+        for (int i = 0; i < oldKeyArray.length; i++) {
+            if (Objects.nonNull(oldKeyArray[i])) {
+                if (Objects.isNull(oldKeyArray[i].next)) {
+                    int newArrayIndex = findIndex(oldKeyArray[i].key);
+                    addNewNode(oldKeyArray[i].key, oldKeyArray[i].value, newArrayIndex);
+                } else {
```

Comment:
add return statement to 'if' block and simplify redundant else condition

Reply:
@al-pavlenko delete redundant bloc of code & add return statement.